airsonic-main/src/main/java/org/airsonic/player/service/metadata/JaudiotaggerParser.java

```diff
@@ -219,42 +219,27 @@ private String mapGenre(String genre) {
      * track numbers on the form "4/12".
      */
     private Integer parseTrackNumber(String trackNumber) {
-        if (trackNumber == null) {
-            return null;
-        }
-
-        Integer result = null;
-
-        try {
-            result = Integer.valueOf(trackNumber);
-        } catch (NumberFormatException x) {
-            Matcher matcher = TRACK_NUMBER_PATTERN.matcher(trackNumber);
-            if (matcher.matches()) {
-                try {
-                    result = Integer.valueOf(matcher.group(1));
-                } catch (NumberFormatException e) {
-                    return null;
-                }
-            }
-        }
-
-        if (Integer.valueOf(0).equals(result)) {
-            return null;
-        }
-        return result;
+        return parseIntegerPattern(trackNumber, TRACK_NUMBER_PATTERN);
     }
 
     private Integer parseYear(String year) {
-        if (year == null) {
+        return parseIntegerPattern(year, YEAR_NUMBER_PATTERN);
+    }
+
+    private Integer parseIntegerPattern(String str, Pattern pattern) {
```

Comment:
Why isn't zero an acceptable return value?

Reply:
This is used for year, tracknumber and disknumber, none of which really make sense to allow zero for.  But mostly I just left it with the same logic and limitations it originally had, which was that they didn't allow zero for those fields.

Reply:
It is better to keep the original.

Airsonic has several null avoidances that are not done in Subsonic.
(Width-Height when Media is VIDEO, PlayCount of Arrist, etc. Unfair null avoidance.)
The part that returns null in the original data specification is almost always an intentional specification.